python/lsst/resources/http.py

```diff
@@ -1325,14 +1325,43 @@ async def get_client_session(**kwargs: Any) -> ClientSession:
                 TCP connections to the server.
             """
             if HttpResourcePath._tcp_connector is None:
-                HttpResourcePath._tcp_connector = TCPConnector(ssl=self._config.ssl_context)
+                HttpResourcePath._tcp_connector = TCPConnector(
+                    # SSL context equipped with client credentials and
+                    # configured to validate server certificates.
+                    ssl=self._config.ssl_context,
+                    # Total number of simultaneous connections this connector
+                    # keeps open with any host.
+                    #
+                    # The default is 100 but we deliberately reduced it to
+                    # avoid keeping a large number of open connexions to file
+                    # servers when thousands of quanta execute simultaneously.
+                    #
+                    # In any case, new connexions are automatically established
+                    # when needed.
+                    limit=10,
```

Comment:
Out of curiosity, are the files spread across multiple HTTP endpoints with different hostnames at FrDF?

Reply:
(Sorry for not responding, I had missed this message)

Yes, there are several front-end hosts behind a DNS load-balancer. In addition, those front-end hosts redirect to the back end server where the requested file is physically located for that server to ultimately serve the request. In our current configuration there are several dozens of back end servers.